### ISSUE TITLE
Add support for projected coordinate systems

### ISSUE THREAD

**Issue by nanshanli** [scikit-mobility/scikit-mobility#185]
Hi,

Firstly, thanks for the great work. I am pleasantly surprised that this module exists and wish I had chanced upon it earlier.

I'd like to suggest adding support for projected coordinate systems in addition to the current WGS84 lat, lon as the world isn't exactly spherical and there are limitations with using the haversine formula to estimate how far & how fast a user is moving. The preprocessing functions depend heavily on these distance and speed calculations.

My proposal for consideration:

* Instead of having `TrajDataFrame` be built on `pandas.DataFrame`, inherit from `geopandas.GeoDataFrame` instead so coordinate reference systems are stored with the dataframe. This makes sense to me since `TrajDataFrame` can be a subset of `GeoDataFrame` (storing only point geometry & with a time element). There are also useful functions in geopandas to convert to various spatial file formats (postGIS, shp, geojson) which could be useful features to add in the future. This could also pave the way for support to upload data that is already in projected coordinate systems (i.e. units of meters rather than degrees).

* The functions within `gislib` will probably need to be modified to detect if the coordinates are projected, or in units of latitude, longitude. If it is projected, Pythagoras Theorem (a^2 + b^2 = c^2) should be used instead of Haversine.

Otherwise, the projection can also be done under the hood (i.e. detect, or specify, suitable projected coordinate system, convert with `pyproj`, use Pythagoras Theorem). 

I am more than happy to contribute to the development of this feature if this is something that resonates with other contributors and users too.

**Comment by miccferr** [comment]
Hi, 

thank you for reaching out! We are glad you find our work useful.

You are raising an interesting point. In the past we decided not to inherit from `geopandas.GeoDataFrame` due to architectural considerations.

Regarding the conversion and re-projection of spatial data, a user might perform these operations as part of pre-processing steps using some excellent libraries e.g. the aforementioned `pyproj`.  

Do you have a specific use case in mind which is not covered by the above?

Thanks again for contacting us.

**Comment by nanshanli** [comment]
Hi -

I was looking into the preprocessing functions in skmob (stop detection, clustering, filtering), which all involve distance-based parameters. The use-cases are not limited to just that module, since any calculation that involves distance can benefit from using projected coordinate systems & the projected cartesian coordinates to compute distance.

From `utils/gislib.py` I observed that the only distance function available is `getDistanceByHaversine`, which assumes the earth is a sphere and uses a mean radius of 6371km to compute distances between 2 (lat, lon) points. 

The main use case I started out with:
* Is there a good way to specify a Pythagorean distance function if I am using a projected coordinate system and use that distance instead of the Haversine distance in the preprocessing functions (and/or other functions in the library) for more accurate parameter specifications?

Addition question / brainstorming:
* It might be useful to have `tdf.crs` return a `pyproj.crs.crs.CRS` object like `geopandas.GeoDataFrame.crs` does, which includes functions like `is_projected` that would be useful in choosing a distance function to use (or what distance functions are not available). 
* A hacky way to start could be to allow users to specify what distance function they'd like to use in the functions (and keep the default as Haversine with (lat, lon) coordinates)
